src/element.js

```diff
@@ -148,38 +157,41 @@ async function elementState(sessionInfo, strategy, selector, driver) {
   return await response.json();
 }
 
-function _setTimeout(
-  elementWaitProps = {
-    timeout: 10000,
-    intervalBetweenAttempts: 500,
-    overrideTimeout: false,
-  },
+function _setPluginProperties(
+  elementWaitProps = defaultTimeOuts,
   session,
   overrideTimeout = false
 ) {
-  if (overrideTimeout && map.has(session)) {
-    map.delete(session);
+  if (!overrideTimeout) {
+    map.set(session, elementWaitProps);
+    log.info(
+      `Timeout properties are set/reset for session ${session} as ${JSON.stringify(
+        _getPluginProperties(session)
+      )}`
+    );
+    return;
   }
-  if (!map.has(session)) {
-    log.info(`Timeout properties not set for session ${session}, trying to set one`);
-    let defaultTimeoutProp = elementWaitProps;
-    if (
-      elementWaitProps.timeout !== undefined &&
-      elementWaitProps.timeout !== defaultTimeoutProp.timeout
-    ) {
-      defaultTimeoutProp.timeout = elementWaitProps.timeout;
-      log.info(`Timeout is set to ${defaultTimeoutProp.timeout}`);
-    }
-    if (
-      elementWaitProps.intervalBetweenAttempts !== undefined &&
-      elementWaitProps.intervalBetweenAttempts !== defaultTimeoutProp.intervalBetweenAttempts
-    ) {
-      defaultTimeoutProp.intervalBetweenAttempts = elementWaitProps.intervalBetweenAttempts;
-      log.info(`Internal between attempts is set to ${defaultTimeoutProp.intervalBetweenAttempts}`);
-    }
-    map.set(session, defaultTimeoutProp);
+  let newtimeouts = Object.assign({}, _getPluginProperties(session));
+  if (elementWaitProps.timeout) {
+    newtimeouts.timeout = elementWaitProps.timeout;
+  }
+  if (elementWaitProps.intervalBetweenAttempts) {
+    newtimeouts.intervalBetweenAttempts = elementWaitProps.intervalBetweenAttempts;
+  }
+  if (elementWaitProps.excludeEnabledCheck) {
+    newtimeouts.excludeEnabledCheck = elementWaitProps.excludeEnabledCheck;
```

Comment:
Can we use object assignment here instead of multiple if statements?
Something like:

const newTimeouts = {
  timeout: elementWaitProps.timeout || defaultTimeOuts.timeout,
  intervalBetweenAttempts: elementWaitProps.intervalBetweenAttempts || defaultTimeOuts.intervalBetweenAttempts,
  excludeEnabledCheck: elementWaitProps.excludeEnabledCheck || defaultTimeOuts.excludeEnabledCheck
};

Reply:
@SrinivasanTarget @saikrishna321 refactored to reduce the complexity of assigning defaults and overrides.